src/plugins/usage_collection/server/plugin.ts

```diff
@@ -154,10 +95,19 @@ export class UsageCollectionPlugin implements Plugin<UsageCollectionSetup> {
 
     this.savedObjects = savedObjects.createInternalRepository();
     if (config.usageCounters.enabled) {
-      this.usageCountersService.start({ savedObjects });
+      const usageCountersStart = this.usageCountersService.start({ savedObjects });
+
+      return {
+        search: usageCountersStart.search,
+      };
     } else {
       // call stop() to complete observers.
       this.usageCountersService.stop();
+      return {
+        search: () => {
+          throw new Error('Usage Counters are not enabled.');
+        },
```

Comment:
nit: how about returning an empty response instead? I wonder if throwing this error will create the need for plugins to check if it's enabled (and we don't provide an API to share if it's enabled or not).

Alternatively, we could allow searching, only that, when `enabled: false`, we don't store more data. WDYT?

Reply:
The problem is I only obtain the search method when calling the `service.start()`. Either:
- I call start (and we'll have a few RxJS timers running without buffering any events)
- Or I make the `search` method public (we break consistency with other services).
- Or I expose the `search` in the response of the `stop()` hook.

Reply:
Fixed in https://github.com/elastic/kibana/pull/187665/commits/f58e23b3394d91bc62962aa6707034e8ea8ae9d1